### PR TITLE
#2650 - Cancel Future Disbursements if COE is declined (Bug)

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
### Student Application in Enrollment status waiting for the COE approval.

image

### Student Application when the first COE is declined, the message for the first COE was adjusted.

image

### Completed Student Application when the second COE is declined.

image

### Student application with only one disbursement declined.

image

### CODE REVIEW

File: sources/packages/backend/libs/services/src/confirmation-of-enrollment/confirmation-of-enrollment.service.ts [hunk 1]
```diff
@@ -424,61 +425,66 @@ export class ConfirmationOfEnrollmentService {
...
-          .createQueryBuilder()
-          .update(DisbursementSchedule)
-          .set({
+    return this.dataSource.transaction(async (transactionalEntityManager) => {
+      const updateResult = await transactionalEntityManager
+        .getRepository(DisbursementSchedule)
+        .update(
+          {
+            coeStatus: COEStatus.required,
+            // Use assessment id to ensure the second disbursement will
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/services/src/confirmation-of-enrollment/confirmation-of-enrollment.service.ts:434]: 👍

File: sources/packages/backend/libs/services/src/confirmation-of-enrollment/confirmation-of-enrollment.service.ts [hunk 2]
```diff
@@ -424,61 +425,66 @@ export class ConfirmationOfEnrollmentService {
...
     const auditUser = { id: auditUserId } as User;
     const now = new Date();
-    return await this.dataSource.transaction(
-      async (transactionalEntityManager) => {
-        const updateResult = await transactionalEntityManager
-          .getRepository(DisbursementSchedule)
-          .createQueryBuilder()
-          .update(DisbursementSchedule)
-          .set({
+    return this.dataSource.transaction(async (transactionalEntityManager) => {
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/services/src/confirmation-of-enrollment/confirmation-of-enrollment.service.ts:428]: Not a blocker or even not part of the PR. `disbursementScheduleId` seems to be obsolete here.

I could see that it is used in the `createNotificationForDisbursementUpdate` but it again uses the disbursement id to get assessment and application and student details.

File: sources/packages/web/src/components/students/applicationTracker/MultipleDisbursementBanner.vue [hunk 3]
```diff
@@ -29,9 +29,14 @@
     background-color="error-bg"
   >
     <template #content
-      ><span class="font-bold">Reason from your institution:</span>
-      {{ coeDenialReason }}. Please note any scheduled payment(s) will be
-      cancelled.</template
+      ><span class="font-bold"
+        >Your institution declined your enrolment status.</span
+      >
+      Reason from your institution: {{ coeDenialReason }}. All disbursements
```
Comment (dheepak-aot) [inline sources/packages/web/src/components/students/applicationTracker/MultipleDisbursementBanner.vue:35]: As discussed, this message can be applicable for single disbursement as well.

Comment (andrewsignori-aot) [reply sources/packages/web/src/components/students/applicationTracker/MultipleDisbursementBanner.vue:35]: Yes, I am checking with business. Thanks for bringing this up.

File: sources/packages/backend/libs/services/src/confirmation-of-enrollment/confirmation-of-enrollment.service.ts (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/services/src/confirmation-of-enrollment/confirmation-of-enrollment.service.ts:428]: If time allows and not many changes are needed I can take a look on it.

**Comment by dheepak-aot** [comment]
Good work @andrewsignori-aot . Just one minor comment.

File: sources/packages/web/src/components/students/applicationTracker/MultipleDisbursementBanner.vue (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/components/students/applicationTracker/MultipleDisbursementBanner.vue]: Update the PR description and send a message to the business for final confirmation on this.

Comment (andrewsignori-aot) [reply sources/packages/web/src/components/students/applicationTracker/MultipleDisbursementBanner.vue]: Business confirmed that we can use the message for the first disbursement in singular, as below.
image
@CarlyCotton FYI

**Review by dheepak-aot — APPROVED** [review]
Thanks for doing the change. Looks good 👍
